packages/server/api/src/app/core/security/authentication-middleware.ts

```diff
@@ -0,0 +1,33 @@
+import { apId, PrincipalType } from '@activepieces/shared'
+import { FastifyRequest } from 'fastify'
+import { apiKeyAuthn } from './authn/api-key-authn'
+import { bearerTokenAuthn } from './authn/bearer-token-authn'
+import { RouteKind } from '@activepieces/server-shared'
+
+export const authenticationMiddleware = async (request: FastifyRequest): Promise<void> => {
+    const security = request.routeOptions.config?.security
+    
+    if (security.kind === RouteKind.PUBLIC) {
+        setAnonymousPrincipal(request)
+        return
+    }
+
+    if (apiKeyAuthn.isApiKey(request)) {
+        request.principal = await apiKeyAuthn.authenticate(request)
+        return
+    }
+
+    if (bearerTokenAuthn.isBearerToken(request)) {
+        request.principal = await bearerTokenAuthn.authenticate(request)
```

Comment:
It's unclear to me when I read this what will happen if the token is invalid, there is no indicator that this function throw error, I would imagine it also can be null.

I think what you did in the engine is great candidate here, where this function expect to return authentication error or data (principal), and rename function to something like verify(request)

First I would understand two things:
- That function is actually verifying the principal.
- In casse it fails It return an error that we will the row again.